redis.conf

```diff
@@ -660,6 +660,18 @@ repl-diskless-sync-max-replicas 0
 #                 during replication.
 repl-diskless-load disabled
 
+# In full sync, Primary can accumulate replication data or send it with the replica rdb,
+# to accumulated there. By enabling the RDB channel (on the replica), the primary's 
+# memory load is moved to the syncing  replica. Additionally, by using rdb channel, the
+# bgsave process streams the snapshot directly to the replica, thereby reducing the CPU
+# load on the primary main process.
+#
+# In primary disk-based sync, RDB channel sync is not permitted because it may cause the
+# primary to start a new sync session for every replica, even if their sync requests are
+# close together.
+#
+repl-rdb-channel no
```

Comment:
Once the feature will be stable, the feature flag can stay on. I prefer removing the feature flag at a later stage.

A delay before starting disk based sync will work. There is a downside of longer syncs, but it is a good workaround.

Regarding issue upon cloning the COB when a new replica joins an existing fork. Do you mean while using disk based sync? And at which step this new replica joins?

Reply:
see copyReplicaOutputBuffer.
i.e. in a disk-based master, a replica that joins an existing fork inherits the COB from another one waiting for the same fork.
this should be impossible now, but we can keep the fork's repl offset in some server var, and check if it happens to still exist in the backlog.

anyway, diskless is the future, so i'm not certain we'd wanna invest here.

Reply:
re-opening this discussion (maybe for a different purpose, or maybe that was your aim all a long..
first i'll state again that the user should never worry about (or be able to) control which mechanism we use for replication.
and if you need some config in order to benchmark things you better just use the old version, or put some ifdef in the code.
however, we do want to write tests that check fallback, surely for a new replica to connect to an old master (required), but maybe also for an old replica to connect to a new master (because we have some 3rd party components acting as fake replicas).
it's best that we write Tcl tests to cover that, and for that we need be able to cause either party to react as if it doesn't understand the new protocol. it's best to do that in a single point (not spread these checks all over), so we should do that at protocol negotiation
for that we can create a config to control it, but then, it should be a HIDDEN config, and it should not be documented in the conf file (just add a single short line in config.c to define it's purpose)

Reply:
OK, good point. I thought we will only need old master to new replica. I will make the config hidden and add some tests.